tiledb/sm/serialization/fragment_info.cc

```diff
@@ -0,0 +1,494 @@
+    // This is needed so that we don't try to load rtee from disk
+    single_frag_info.meta()->set_rtree_loaded();
+    // set the rest of single_frag_info from meta()
+    single_frag_info.set_info_from_meta();
+  } else {
+    return {
+        Status_SerializationError(
+            "Missing fragment metadata from single fragment info capnp reader"),
+        nullopt};
+  }
+
+  // Get fragment size
+  single_frag_info.fragment_size() = single_frag_info_reader.getFragmentSize();
+
+  return {Status::Ok(), single_frag_info};
+}
+
+Status single_fragment_info_to_capnp(
+    const SingleFragmentInfo& single_frag_info,
+    capnp::SingleFragmentInfo::Builder* single_frag_info_builder) {
+  // set array schema name
+  single_frag_info_builder->setArraySchemaName(
+      single_frag_info.array_schema_name());
+
+  // set fragment metadata
+  auto frag_meta_builder = single_frag_info_builder->initMeta();
+  RETURN_NOT_OK(
+      fragment_metadata_to_capnp(*single_frag_info.meta(), &frag_meta_builder));
+
+  // set fragment size
+  single_frag_info_builder->setFragmentSize(single_frag_info.fragment_size());
+  return Status::Ok();
+}
+
+Status fragment_info_from_capnp(
+    const capnp::FragmentInfo::Reader& fragment_info_reader,
+    const URI& array_uri,
+    FragmentInfo* fragment_info) {
+  // Get array_schemas_all from capnp
+  if (fragment_info_reader.hasArraySchemasAll()) {
+    auto all_schemas_reader = fragment_info_reader.getArraySchemasAll();
+
+    if (all_schemas_reader.hasEntries()) {
+      auto entries = fragment_info_reader.getArraySchemasAll().getEntries();
+      for (auto array_schema_build : entries) {
+        auto schema{
+            array_schema_from_capnp(array_schema_build.getValue(), array_uri)};
+        fragment_info->array_schemas_all()[array_schema_build.getKey()] =
+            make_shared<ArraySchema>(HERE(), schema);
+      }
+    }
+  }
+
+  // Get single_fragment_info from capnp
+  if (fragment_info_reader.hasFragmentInfo()) {
+    for (auto single_frag_info_reader :
+         fragment_info_reader.getFragmentInfo()) {
+      auto&& [st, single_frag_info] = single_fragment_info_from_capnp(
+          single_frag_info_reader, fragment_info->array_schemas_all());
+      RETURN_NOT_OK(st);
+      fragment_info->single_fragment_info_vec().emplace_back(
+          single_frag_info.value());
+    }
+  }
+
+  // Get uris to vacuum from capnp
+  if (fragment_info_reader.hasToVacuum()) {
+    for (auto uri : fragment_info_reader.getToVacuum()) {
+      fragment_info->to_vacuum().emplace_back(uri.cStr());
+    }
+  }
+
+  // Fill in the rest of values of fragment info class using the above
+  // deserialized values
+  fragment_info->array_uri() = array_uri;
+  fragment_info->unconsolidated_metadata_num() = 0;
+  for (const auto& f : fragment_info->single_fragment_info_vec()) {
+    fragment_info->unconsolidated_metadata_num() +=
+        (uint32_t)!f.has_consolidated_footer();
+  }
+  // TODO: do we need anterior_ndrange_?
```

Comment:
Whilst it doesn't seem used in the fragment_info code itself, it seems to be used in the fragment consolidator, also it doesn't seem reconstructed anywhere in the fragment_info code, I would assume this needs reconstruction here as well.